Samples/NetworkMediaStreamer/CameraRTPStreamerApp/CameraRTPStreamerApp.cpp

```diff
@@ -0,0 +1,419 @@
+    check_hresult(MFSetAttributeSize(spInType.get(), MF_MT_FRAME_SIZE, width, height));
+    check_hresult(MFSetAttributeRatio(spInType.get(), MF_MT_FRAME_RATE, format.FrameRate().Numerator(), format.FrameRate().Denominator()));
+
+    check_hresult(spInType->SetGUID(MF_MT_MAJOR_TYPE, MFMediaType_Video));
+    check_hresult(spInType->SetGUID(MF_MT_SUBTYPE, subType ));
+
+    check_hresult(MFCreateAttributes(spSWAttributes.put(), 2));
+    check_hresult(spSWAttributes->SetUINT32(MF_LOW_LATENCY, TRUE));
+    HRESULT hr = S_OK;
+    BOOL bEnableHWTransforms = FALSE;
+    do
+    {
+        spSinkWriter = nullptr;
+        check_hresult(spSWAttributes->SetUINT32(MF_READWRITE_ENABLE_HARDWARE_TRANSFORMS, bEnableHWTransforms));
+        check_hresult(MFCreateSinkWriterFromMediaSink(pMediaSink, spSWAttributes.get(), spSinkWriter.put()));
+        hr = spSinkWriter->SetInputMediaType(0, spInType.get(), nullptr);
+        bEnableHWTransforms = !bEnableHWTransforms;
+    } while (hr == MF_E_TOPO_CODEC_NOT_FOUND);
+    check_hresult(hr);
+    check_hresult(spSinkWriter->BeginWriting());
+
+    return spSinkWriter.detach();
+}
+#endif
+
+PROCESS_INFORMATION StartLoggerConsole(std::wstring filename)
+{
+        STARTUPINFO si,si1;
+        PROCESS_INFORMATION pi;
+        GetStartupInfo(&si1);
+        ZeroMemory(&si, sizeof(si));
+        si.cb = sizeof(si);
+        ZeroMemory(&pi, sizeof(pi));
+        std::wstring cmdline = L"powershell Get-Content " + filename + L" -Wait ";
+        std::wstring title = L"RTSPServer Log Console";
+        si.lpTitle = (LPWSTR)title.c_str();
+        
+        si.dwX = si1.dwX + 500;
+        si.dwY = si1.dwY;
+        si.dwXCountChars = 40;
+        si.dwYCountChars = 10;
+        si.dwFlags = STARTF_USESHOWWINDOW | STARTF_USEPOSITION | STARTF_PREVENTPINNING | STARTF_USECOUNTCHARS;
+        si.wShowWindow = SW_SHOWNOACTIVATE;// | SW_HIDE;
+        // Start the child process. 
+        if (!CreateProcess(NULL,   // No module name (use command line)
+            (LPWSTR)cmdline.c_str(),        // Command line
+            NULL,           // Process handle not inheritable
+            NULL,           // Thread handle not inheritable
+            FALSE,          // Set handle inheritance to FALSE
+            CREATE_NEW_CONSOLE,              // No creation flags
+            NULL,           // Use parent's environment block
+            NULL,           // Use parent's starting directory 
+            &si,            // Pointer to STARTUPINFO structure
+            &pi)           // Pointer to PROCESS_INFORMATION structure
+            )
+        {
+            check_win32(GetLastError());
+        }
+        return pi;
+    
+}
+
+void StopLoggerConsole(PROCESS_INFORMATION pi)
+{
+    TerminateProcess(pi.hProcess, 0);
+    // Close process and thread handles. 
+    CloseHandle(pi.hProcess);
+    CloseHandle(pi.hThread);
+}
+int main()
+{
+    std::cout << "VideoSteamer App \n";
+    PROCESS_INFORMATION loggerConsole;

```

Comment:
>PROCESS_INFORMATION [](start = 4, length = 19)

suggestion to use wil::unique_process_information for automatic cleanup

Reply:
i cannot use wil as it needs nuget.  This code will eventually go to ROS and using nugets in cmake is painful. 

---
In reply to: [491091708](https://github.com/microsoft/Windows-Camera/pull/13#discussion_r491091708) [](ancestors = 491091708)